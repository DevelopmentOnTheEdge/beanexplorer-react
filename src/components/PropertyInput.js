import React                from 'react';
import PropTypes            from 'prop-types';
import Datetime             from 'react-datetime';
import moment               from 'moment';
import Select               from 'react-select';
import VirtualizedSelect    from 'react-virtualized-select'
import NumericInput          from 'react-numeric-input';
import CKEditor              from 'react-ckeditor-component';
import MaskedInput           from 'react-maskedinput';
import JsonPointer          from 'json-pointer';


class PropertyInput extends React.Component
{
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
  }

  getPath() {
    if(this.props.path) {
      return this.props.path;
    }else{
      return this.props.bean.order[this.props.id];
    }
  }

  callOnChange(value) {
    this.props.onChange(this.getPath(), value);
  }

  handleChange(event) {
    const value = (event.target.type === 'checkbox') ? event.target.checked : event.target.value;
    this.callOnChange(value);
  }

  dateToISOFormat(date) {
    if(typeof date === "string") {
      this.callOnChange(date);
    } else {
      this.callOnChange(date.format('YYYY-MM-DD'));
    }
  }

  dateFromISOFormat(stringDate) {
    const date = moment(stringDate === undefined ? "" : stringDate, 'YYYY-MM-DD', true);
    if (date.isValid()) {
      return date.format('DD.MM.YYYY');
    } else {
      return stringDate;
    }
  }

  handleChangeSelect(object) {
    if(Array.isArray(object)) {
      let selectArray = [];
      Object.keys(object).forEach(function (key) {
        selectArray.push(object[key].value);
      });
      this.callOnChange(selectArray);
    } else {
      this.callOnChange(object !== null ? object.value : "");
    }
  }

  static getMaskInput(rules)
  {
    for (let i = 0; i < rules.length; i++)
    {
      if ("mask" in rules[i]) {
        return rules[i].mask
      }
    }
    return null;
  }

  static isNumberInput(rules)
  {
    for (let i =0 ; i< rules.length; i++)
    {
      if(rules[i].type === "baseRule" &&
        ( rules[i].attr === "digits" || rules[i].attr === "integer" || rules[i].attr === "number" ))return true;
    }
    return false;
  }

  static getNumericProps(meta)
  {
    let props = {};
    props['maxLength'] = 14;//errors if more
    const rules = meta.validationRules;
    for (let i =0 ; i< rules.length; i++)
    {
      if(rules[i].type === "baseRule" && (rules[i].attr === "number"))
      {
        props['precision'] = 10;
      }
      if(rules[i].type === "baseRule" && (rules[i].attr === "integer"))
      {
        props['min'] = -2147483648;
        props['max'] = 2147483647;
        props['maxLength'] = 9;
        props['precision'] = 0;
      }
      // if(rules[i].type === "digits")
      // {
      //   props['min'] = 0;//todo not work
      // }
    }
    if(meta.columnSize){
      props['maxLength'] = parseInt(meta.columnSize);
    }
    return props;
  }

  static getBase64(file) {
    return new Promise((resolve, reject) => {
      const reader = new FileReader();
      reader.addEventListener("load", function () {
        resolve(reader.result);
      }, false);
      reader.onerror = error => reject(error);

      reader.readAsDataURL(file);
    });
  }

  static getExtraAttrsMap(extraAttrs){
    let map = {};
    if(extraAttrs === undefined)return map;
    for (let i=0 ;i< extraAttrs.length; i++){
      map[extraAttrs[i][0]] = extraAttrs[i][1];
    }
    return map;
  }

  render() {
    const path  = this.getPath();
    const meta  = this.props.bean.meta[path];
    const value = JsonPointer.get(this.props.bean, "/values" + path);
    const id    = path.substring(path.lastIndexOf("/")+1) + "PropertyInput";
    const extraAttrsMap = PropertyInput.getExtraAttrsMap(meta.extraAttrs);
    const required = meta.canBeNull !== true;

    const baseProps = {
      id: id,
      key: id,
      disabled: meta.readOnly,
      required: required
    };

    const rawTextInputProps = Object.assign({}, baseProps, {
      value: value === undefined ? "" : value,
      onChange: this.handleChange,
      placeholder: meta.placeholder,
      className: this.props.controlClassName || "form-control"
    });

    const controls = {
      textInput: () => (
        <input type="text" {...rawTextInputProps} />
      ),
      passwordField: () => (
        <input type="password" {...rawTextInputProps} />
      ),
      textArea: () => (
        <textarea rows={meta.rows || 3} cols={meta.columns} {...rawTextInputProps} />
      ),
      Boolean: () => (
        <input type="checkbox" checked={value === true || value === "true"} onChange={this.handleChange}
               className={this.props.controlClassName || 'form-check-input'} {...baseProps} />
      ),
      file: () => (
        <input type="file" className={this.props.controlClassName || "form-control-file"} {...baseProps}
                      multiple={meta.multipleSelectionList}
                      onChange={(e) => {
                        if(e.target.files && e.target.files.length === 1) {
                          const fileName = e.target.files[0].name;
                          PropertyInput.getBase64(e.target.files[0]).then(data => {
                            this.callOnChange({type: "Base64File", name: fileName, data: data})
                          });
                        }else if(e.target.files && e.target.files.length === 0) {
                          this.callOnChange("")
                        }
                      }} />
      ),
      select: () => {
        let options = [];
        for(let i =0 ;i < meta.tagList.length; i++){
          options.push({ value: meta.tagList[i][0], label: meta.tagList[i][1] });
        }

        let strValue;
        if(Array.isArray(value)){
          strValue = [];
          for (let i = 0; i < value.length; i++)strValue.push("" + value[i]);
        }
        else
        {
          strValue = "" + value;
        }
        const selectAttr = {
          ref: id, name: id, value: strValue, options: options, onChange: (v) => this.handleChangeSelect(v),
          clearAllText: this.props.localization.clearAllText,
          clearValueText: this.props.localization.clearValueText,
          noResultsText: this.props.localization.noResultsText,
          searchPromptText: this.props.localization.searchPromptText,
          loadingPlaceholder: this.props.localization.loadingPlaceholder,
          placeholder: meta.placeholder || this.props.localization.placeholder,
          backspaceRemoves: false,
          disabled: meta.readOnly,
          multi: meta.multipleSelectionList,
          matchPos: extraAttrsMap.matchPos || "any",
          required: required
        };

        if(extraAttrsMap.inputType === "Creatable")
        {
          return <Creatable {...selectAttr} />
        }
        else if(extraAttrsMap.inputType === "VirtualizedSelect")
        {
          return <VirtualizedSelect {...selectAttr} clearable searchable labelKey="label" valueKey="value" />
        }
        else
        {
          return <Select {...selectAttr} />
        }
      },
      Date: () => (
        <Datetime dateFormat="DD.MM.YYYY" id={id} key={id} inputProps={ {disabled: meta.readOnly, required: required} }
                         onChange={(v) => this.dateToISOFormat(v)} value={this.dateFromISOFormat(value)}
                         timeFormat={false} closeOnSelect={true} closeOnTab={true} locale={this.props.localization.locale || "en"} />
      ),
//      dateTime: {
//        normal: () => {
//          return ( React.createElement(Datetime, {id: id, key: id, value: value, parent: _this, onChange: handleChange, time: true, className: this.props.controlClassName}) );
//        },
//        readOnly: () => this.createStatic(value)
//      },
      maskTest: () => (
        <MaskedInput mask={PropertyInput.getMaskInput(meta.validationRules)} value={value === undefined ? "" : value}
                            onChange={this.handleChange} className={this.props.controlClassName || "form-control"} {...baseProps} />
      ),
      numberInput: () => {
        const numericProps = PropertyInput.getNumericProps(meta);
        return <NumericInput {...numericProps} placeholder={meta.placeholder} value={value} {...baseProps}
                             onChange={(valueAsNumber, valueAsString, input) => {
                               this.callOnChange(valueAsNumber !== null ? valueAsNumber : "");
                             }}
                             style={ false } className={this.props.controlClassName || "form-control"} />
      },
      WYSIWYG: () => (
        <CKEditor activeClass="p10" content={value}
                         events={{
                           "change": (evt) => { this.callOnChange(evt.editor.getData()) }
                         }}
                         config={{language: 'ru', readOnly: meta.readOnly}}
        />
      ),
      labelField: () => {
        if(meta.rawValue)
        {
          return <div dangerouslySetInnerHTML={{__html: value}} />
        }
        else
        {
          return <label className="form-control-label">{value}</label>
        }
      },
    };

    if(meta.tagList)
    {
      return controls['select']();
    }

    if(meta.passwordField)
    {
      return controls['passwordField']();
    }

    if(meta.labelField)
    {
      return controls['labelField']();
    }

    if(meta.validationRules !== undefined && PropertyInput.isNumberInput(meta.validationRules))
    {
      return controls['numberInput']()
    }

    if(extraAttrsMap.inputType === 'WYSIWYG')
    {
      return controls['WYSIWYG']();
    }

    if(extraAttrsMap.inputType === 'textArea')
    {
      return controls['textArea']();
    }

    if(extraAttrsMap.inputType === 'file')
    {
      return controls['file']();
    }

    if(meta.validationRules !== undefined && PropertyInput.getMaskInput(meta.validationRules))
    {
      return controls['maskTest']();
    }

    if(controls[meta.type] !== undefined)
    {
      return controls[meta.type]();
    }

    return controls['textInput']();
  }

}

PropertyInput.defaultProps = {
  localization: {
    locale: 'en',
    clearAllText: 'Clear all',
    clearValueText: 'Clear value',
    noResultsText: 'No results found',
    searchPromptText: 'Type to search',
    placeholder: 'Select ...',
    loadingPlaceholder: 'Loading...'
  },
};

PropertyInput.propTypes = {
  bean: PropTypes.object.isRequired,
  path: PropTypes.string,
  id: PropTypes.number,
  onChange: PropTypes.func,
  localization: PropTypes.object,
  controlClassName: PropTypes.string, 
};

export default PropertyInput;
