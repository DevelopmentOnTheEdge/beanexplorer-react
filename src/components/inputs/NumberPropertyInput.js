import React from 'react';
import bigInt from "big-integer";
import bigRat from "big-rational";
import BasePropertyInput from "./BasePropertyInput";


export default class NumberPropertyInput extends BasePropertyInput {
  constructor(props) {
    super(props);
    this.numberValidation = this.numberValidation.bind(this);
  }

  render() {
    const meta = this.getMeta();
    const extraAttrsMap = BasePropertyInput.getExtraAttrsMap(meta);
    const range = this.getNumberValidationRule('range');
    const step = this.getNumberValidationRule('step');
    const type = meta.type;

    return <input
      type="text"
      onInput={this.numberValidation}
      data-info-type={type}
      data-info-range={(range && range.attr) ? range.attr.min + ', ' + range.attr.max : undefined}
      data-info-step={step ? step.attr : undefined}
      value={convertENotationNumbers(this.getValue())}
      onChange={this.handleChange}
      placeholder={extraAttrsMap.placeholder}
      {...this.getBaseProps()}
    />
  }

  numberValidation(e) {
    const range = this.getNumberValidationRule('range');
    const step = this.getNumberValidationRule('step');

    const local = this.props.localization;

    let value;
    try {
      value = bigRat(e.target.value);
    } catch (err) {
      setErrorState(e, local.numberTypeMismatch);
      return
    }

    if (range) {
      if (value.compare(bigRat(range.attr.min)) === -1) {
        setErrorState(e, setMessagePlaceHolders(local.rangeUnderflow, [range.attr.min]));
        return
      }
      else if (value.compare(bigRat(range.attr.max)) === 1) {
        setErrorState(e, setMessagePlaceHolders(local.rangeOverflow, [range.attr.max]));
        return
      }
    }

    if (step) {
      const stepRat = bigRat(step.attr);

      if (!value.divide(stepRat).denominator.equals(bigInt.one)) {
        const min = value.divide(stepRat).floor().multiply(stepRat);
        const max = min.add(stepRat);

        setErrorState(e, setMessagePlaceHolders(
          local.stepMismatch, [min.toDecimal(), max.toDecimal()]
        ));
        return
      }
    }

    setErrorState(e, '');
  }

  getNumberValidationRule(name) {
    const meta = this.getMeta();
    let rule = this.getValidationRule(name);

    if (rule === undefined)
    {
      if (name === 'range') {
        switch (meta.type) {
          case 'Short':
            return {type: 'range', attr: {min: "-32768", max: "32767"}};
          case 'Integer':
            return {type: 'range', attr: {min: "-2147483648", max: "2147483647"}};
          case 'Long':
            return {type: 'range', attr: {min: "-9223372036854775808", max: "9223372036854775807"}};
        }
      }
      if (name === 'step' && (meta.type === 'Short' || meta.type === 'Integer' || meta.type === 'Long')) {
        return {type: 'step', attr: '1'};
      }
    }
    return rule;
  };
}

const convertENotationNumbers = function (value) {
  try {
    if (value.includes('e') || value.includes('E'))
      return bigRat(value).toDecimal();
    else
      return value;
  } catch (err) {
    return value
  }
};

const setErrorState = function (e, text) {
  e.target.setCustomValidity(text);
  e.target.title = text;
};

const setMessagePlaceHolders = function (source, params) {
  params.forEach(function (item, i) {
    source = source.replace(new RegExp("\\{" + i + "\\}", "g"), item);
  });
  return source
};
