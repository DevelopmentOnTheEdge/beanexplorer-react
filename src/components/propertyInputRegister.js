const propertyInputs = {};

export const getPropertyInput = (name) => {
  return propertyInputs[name];
};

export const registerPropertyInput = (name, fn) => {
  propertyInputs[name] = fn;
};

export const getAllPropertyInputs = () => {
  return Object.keys(propertyInputs);
};
