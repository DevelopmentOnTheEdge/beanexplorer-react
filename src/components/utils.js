import classNames from 'classnames';

export const inputLabelSizeClasses = (props) => {
  return classNames(
    {'col-form-label-sm': props.bsSize === "sm"},
    {'col-form-label-lg': props.bsSize === "lg"}
  )
};

export const arraysEqual = function (a, b) {
  if (a === b) return true;
  if (a === null || b === null) return false;
  if (a.length !== b.length) return false;

  a.sort();
  b.sort();

  for (let i = 0; i < a.length; ++i) {
    if (a[i] !== b[i]) return false;
  }
  return true;
};

export const shouldPropertyUpdate = function (props, nextProps) {
  return props.bean !== nextProps.bean
    || props.horizontal !== nextProps.horizontal
    || props.inline !== nextProps.inline
    || props.bsSize !== nextProps.bsSize;
};

//experemental, try check nested DPS
export const isDPS = function (meta) {
  return meta && meta.type && meta.type.indexOf('DynamicPropertySet') !== -1
}