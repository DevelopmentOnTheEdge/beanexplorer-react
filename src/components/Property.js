import React         from 'react';
import PropTypes     from 'prop-types';
import classNames    from 'classnames';
import PropertyInput from './PropertyInput'


class Property extends React.Component
{
  getPath(){
    if(this.props.path){
      return this.props.path;
    }else{
      return this.props.bean.order[this.props.id];
    }
  }

  render() {
    const path = this.getPath();
    const meta  = this.props.bean.meta[path];
    const id    = path.substring(path.lastIndexOf("/")+1) + "PropertyInput";

    const label = <label htmlFor={id} className={meta.type === "Boolean" ? 'form-check-label' : 'form-control-label'}>
                         {meta.displayName || id}</label>;

    const messageElement = meta.message ? <small className={this.props.messageClassName || "form-control-feedback"}>{meta.message}</small> : undefined;

    let hasStatusClasses = classNames(
      {'has-error' : meta.status === 'error'},
      {'has-warning' : meta.status === 'warning'},
      {'has-success' : meta.status === 'success'},
    );

    const classNameForm = (meta.type === "Boolean")
      ? this.props.classNameFormCheck || 'form-check property'
      : this.props.classNameFormGroup || 'form-group property';

    const cssClasses = meta.cssClasses || 'col-lg-12';

    const outerClasses = classNames(
      cssClasses,
      {'display-none' : meta.hidden}
    );

    const classes = classNames(
      classNameForm,
      hasStatusClasses,
      {'required' : meta.canBeNull !== true}
    );

    if(this.props.inline)
    {
      if(meta.type === "Boolean")
      {
        return (
          <div className="form-check mb-2 mr-sm-2">
            <PropertyInput {...this.props} />
            {label}
          </div>
        );
      }
      else
      {
        return (
          <PropertyInput {...this.props} controlClassName="mb-2 mr-sm-2"/>
        );
      }
    }
    else
    {
      if(meta.type === "Boolean")
      {
        return (
          <div className={outerClasses}>
            <div className={classes}>
              <PropertyInput {...this.props} />
              {label}
              {messageElement}
            </div>
          </div>
        );
      }
      else if(meta.labelField)
      {
        return (
          <div className={classNames('form-group property property-label', meta.cssClasses || 'col-lg-12', hasStatusClasses)}>
            <PropertyInput {...this.props} />
          </div>
        );
      }
      else
      {
        return (
          <div className={outerClasses}>
            <div className={classes}>
              {label}
              <div className="controls">
                <PropertyInput {...this.props} />
                {messageElement}
              </div>
            </div>
          </div>
        );
      }
    }

  }

}

Property.defaultProps = {
  localization: {
    locale: 'en',
    clearAllText: 'Clear all',
    clearValueText: 'Clear value',
    noResultsText: 'No results found',
    searchPromptText: 'Type to search',
    placeholder: 'Select ...',
    loadingPlaceholder: 'Loading...'
  },
};

Property.propTypes = {
  bean: PropTypes.object.isRequired,
  path: PropTypes.string,
  id: PropTypes.number,
  inline: PropTypes.bool,
  onChange: PropTypes.func,
  localization: PropTypes.object
};

export default Property;
