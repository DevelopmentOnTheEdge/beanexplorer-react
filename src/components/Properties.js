import React from 'react';
import PropTypes from 'prop-types';
import Property from './Property';
import JsonPointer from 'json-pointer';

class Properties extends React.Component {

  shouldComponentUpdate(nextProps) {
    return this.props.bean !== nextProps.bean || this.props.values !== nextProps.values;
  }

  render() {
    let fields = this.props.bean.order.map((path, i) => {
      if (this.props.ids === undefined || this.props.ids.includes(i)) {
        return (
          <Property path={path} key={path} {...this.props} value={this.getValue(path)}/>
        )
      } else {
        return null;
      }
    });

    //todo remove outer element after migrate to react 16.2
    return <div className={this.props.rowClass}>{fields}</div>;
  }

  getValue(path) {
    const values = this.props.values || this.props.bean.values;
    return JsonPointer.get(values, path)
  }
}

Properties.defaultProps = {
  rowClass: "form-row",
};

Properties.propTypes = {
  rowClass: PropTypes.string,
  bean: PropTypes.object.isRequired,
  ids: PropTypes.array,
  inline: PropTypes.bool,
  bsSize: PropTypes.string,
  onChange: PropTypes.func,
  reloadOnChange: PropTypes.func,
  localization: PropTypes.object
};

export default Properties;
