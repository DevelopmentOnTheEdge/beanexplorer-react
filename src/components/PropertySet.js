import React from 'react';
import PropTypes from 'prop-types';
import Property from './Property';
import classNames from 'classnames';
import JsonPointer from 'json-pointer';
import {shouldPropertyUpdate} from "./utils";

class PropertySet extends React.Component {

  shouldComponentUpdate(nextProps) {
    return shouldPropertyUpdate(this.props, nextProps) || this.props.values !== nextProps.values;
  }

  static getName(name) {
    if (name) {
      return <h5 className='property-group__title'>{name}</h5>
    } else {
      return null
    }
  }

  createGroup(curGroup, curGroupId, curGroupName, curGroupClasses) {
    return (
      <div
        className={classNames(
          'property-group',
          curGroupClasses || 'property-group__top-line col-12'
        )}
        key={curGroupId}
        ref={curGroupId}
      >
        <div className="property-group__top-line-row row"/>
        {PropertySet.getName(curGroupName)}
        <div className={classNames('property-group__row', this.props.rowClass)}>
          {curGroup}
        </div>
      </div>
    );
  }

  render() {
    let curGroup = [];
    let curGroupName = null, curGroupId = null, curGroupClasses = null;
    let fields = [];

    const finishGroup = () => {
      if (curGroup.length > 0) {
        if (curGroupId) {
          fields.push(this.createGroup(curGroup, curGroupId, curGroupName, curGroupClasses));
        } else {
          Array.prototype.push.apply(fields, curGroup);
        }
      }
      curGroup = [];
    };

    for (const path of this.props.bean.order) {
      const meta = this.props.bean.meta[path];

      const newGroupId = meta.groupId;
      const newGroupName = meta.groupName;
      const newGroupClasses = meta.groupClasses;

      if (newGroupId !== curGroupId) {
        finishGroup();
        curGroupName = newGroupName;
        curGroupClasses = newGroupClasses;
        curGroupId = newGroupId;
      }

      const field = (
        <Property key={path} path={path} {...this.props} value={this.getValue(path)}/>
      );

      curGroup.push(field);
    }
    finishGroup();

    return (
      <div className={classNames('property-set', this.props.rowClass)}>
        {fields}
      </div>
    );
  }

  getValue(path) {
    const values = this.props.values || this.props.bean.values;
    return JsonPointer.get(values, path)
  }

}

PropertySet.defaultProps = {
  rowClass: 'form-row'
};

PropertySet.propTypes = {
  bean: PropTypes.object.isRequired,
  values: PropTypes.object,
  onChange: PropTypes.func,
  inline: PropTypes.bool,
  horizontal: PropTypes.bool,
  bsSize: PropTypes.string,
  localization: PropTypes.object,
  rowClass: PropTypes.string
};

export default PropertySet;
