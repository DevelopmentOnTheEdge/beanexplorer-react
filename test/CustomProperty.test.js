import React from 'react';
import Property from '../src/components/Property';
import renderer from 'react-test-renderer';
import NumberPropertyInput from "../src/components/inputs/NumberPropertyInput";
import {registerPropertyInput, getAllPropertyInputs} from "../src/components/propertyInputRegister";


class NumberInputGroup extends NumberPropertyInput {
  render () {
    return (
      <div>
        <div className="input-group mb-3">
          <div className="input-group-prepend">
            <span className="input-group-text">$</span>
          </div>
          {super.render()}
          <div className="input-group-append">
            <span className="input-group-text">.00</span>
          </div>
        </div>
      </div>
    )
  }
}

test('get by id', () => {
  registerPropertyInput('NumberInputGroup', NumberInputGroup);
  const bean = {
    "values": { "input": "no date" },
    "meta":   { "/input": {"type": "Date", "extraAttrs": [["inputType", "NumberInputGroup"]]} },
    "order":  [ "/input" ]
  };

  const component = renderer.create(
    <Property id={0} bean={bean} onChange={undefined} value={"test"}/>
  );
  expect(component.toJSON()).toMatchSnapshot();
  expect(getAllPropertyInputs()).toEqual(['NumberInputGroup']);
});
