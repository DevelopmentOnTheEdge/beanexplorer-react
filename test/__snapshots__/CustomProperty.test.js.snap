// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get by id 1`] = `
<div
  className="vertical-input col-lg-12"
>
  <div
    className="property form-group required"
  >
    <div
      className="controls"
    >
      <div>
        <div
          className="input-group mb-3"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
            >
              $
            </span>
          </div>
          <input
            className="property-input form-control"
            data-info-range={undefined}
            data-info-step={undefined}
            data-info-type="Date"
            id="inputPropertyInput"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            placeholder={undefined}
            required={true}
            size={undefined}
            type="text"
            value="test"
          />
          <div
            className="input-group-append"
          >
            <span
              className="input-group-text"
            >
              .00
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
